python/cudf/cudf/dataframe/string.py

```diff
@@ -529,7 +529,7 @@ def element_indexing(self, arg):
         else:
             raise NotImplementedError(type(arg))
 
-        if len(out) == 1:
+        if isinstance(arg, Number):
```

Comment:
Instead of checking this again here, we can just add the return statement to the original `isinstance(arg, Number)` check above after `out` is defined. Then remove the else condition below and just have it always return the `columnops.as_column(out)`

Reply:
Bumped it up into the existing condition for scalars.